### ISSUE TITLE
How to use this library

### ISSUE THREAD

**Issue by eugyev** [handsontable/react-handsontable#1]
Is there any documentation on how to use this with handsontable pro?

**Comment by jansiegel** [comment]
@eugyev We're still working on finishing the wrapper. After it's done, we're gonna post a proper `README` and a more understandable demo page, so there'll be no misconceptions on how to use it with either Handsontable or Handsontable Pro.
Thanks for checking out `react-handsontable` by the way!

**Comment by eugyev** [comment]
Thanks for the input.  I actually got pretty far using files from this repo already, so it's a great start.  I basically just utilized the HotTable component and the handsontableSettingsMapper in my project directly, and it's working so far.  Getting it to play nicely with redux and meteor is another issue - especially when adding/removing columns etc. but seems like the settingsMapper was the most important piece.

**Comment by jansiegel** [comment]
@eugyev We've prepared an installation guide for this component. You can take a look at it here: https://github.com/handsontable/react-handsontable/wiki/Installation-guide Does it clarify the installation process for Handsontable Pro for you?

**Comment by sandys** [comment]
@eugyev  @jansiegel can someone help on what is the correct pattern to use with redux or mobx ? I'm a little confused on how to refresh data, work with data, etc in the right way with the flux/redux/mobx mechanism.
The single biggest challenge is how to handle callbacks (where to define them in context of react, etc)

This would be very very helpful! We are a Pro member and love handsontable-pro

**Comment by jansiegel** [comment]
@sandys I've prepared a simple "change log" app with `react-handsontable` and Redux: 
http://codepen.io/handsoncode/pen/LWmvPX?editors=0010

Basically, I'm using the `afterChange` hook defined in the `HotTable` configuration to update the Redux store state, and then display it as a list in another component. 
Does this help?

**Comment by sandys** [comment]
Hi Jan
First - thanks for this. I really appreciate your help.

This is helpful, but most people use HOT alone - it means that:
1. Change something in hot
2. Redux does something with it..maybe calls an api.
3. The data gets updated (some columns are computed on the basis of others)
in HOT itself.

A very classic spreadsheet like behavior with some computations happening
in redux/server-side. This is the most popular usecase of hot that I have
seen and used.

Can you talk about the pattern here ?

On Mar 23, 2017 20:14, "Jan Siegel" <notifications@github.com> wrote:

@sandys <https://github.com/sandys> I've prepared a simple "change log" app
with react-handsontable and Redux:
http://codepen.io/handsoncode/pen/LWmvPX?editors=0010

Basically, I'm using the afterChange hook defined in the HotTable
configuration to update the Redux store state, and then display it as a
list in another component.
Does this help?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/handsontable/react-handsontable/issues/1#issuecomment-288741475>,
or mute the thread

**Comment by jdpagley** [comment]
@sandys did you ever figure out how to use hot directly inside of react?

 I'm trying to implement this example http://jsfiddle.net/c9xk156b/ to retrieve row data on the afterChange hook but I need access to hot.

**Comment by nadim** [comment]
Are the wiki instructions for using this library current and expected to work? I am having trouble with both handsontable and handontable-pro. I followed the wiki and end up with this error:
```
Uncaught Error: Cannot find module 'numbro'
```
or
```
Uncaught Error: Cannot find module 'pikaday'
```

I have updated my webpack config as explained in the wiki + installed the appropriate modules.

**Comment by laura-rodriguez** [comment]
Hi,

I am starting a new project and we are considering handsontable-pro. 
I need to display a gantt, but I am not sure how to do this with react-handsontable. Do you have an example for this?

Thanks!

**Comment by jansiegel** [comment]
@sandys You'd probably want to bind the redux state with HotTable's props. 

In this example, before each change I'm dispatching information about the change to the redux store. The redux store updates its state, which triggers HotTable's update, as the store state is bound to the `cells` property. This results in highlighting the last edited cell. You can as well bind the store state to `data` or `columns` props to update any Handsontable config options.
https://www.webpackbin.com/bins/-KpJs_3ncqtW8FRRe2IC

**Comment by jansiegel** [comment]
@jdpagley You don't need to reference `hot` to access the Handsontable instance inside the hook callback. You can simply use `this`, as the callbacks are triggered in Handsontable core context.

Take a look at this example: https://www.webpackbin.com/bins/-KpJzJX662i0zOh_t4mo

**Comment by jansiegel** [comment]
@laura-rodriguez It should work exactly as any other configuration of Handsontable, by simply passing the Gantt Chart plugin configuration to the React component's props:
```js
this.dataObj = [
  {
    startDate: '1/5/2015',
    endDate: '1/20/2015',
    additionalData: {label: 'Label 1'}
  },
  {
    startDate: '1/11/2015',
    endDate: '1/29/2015',
    additionalData: {label: 'Label 2'}
  },
  {
    startDate: '2/1/2015',
    endDate: '2/26/2015',
    additionalData: {label: 'Label 3'}
  },
  {
    startDate: '2/15/2015',
    endDate: '3/26/2015',
    additionalData: {label: 'Label 4'}
  }
];
```

```jsx
<HotTable
  ganttChart={{
    dataSource: this.dataObj,
    firstWeekDay: 'monday',
    startYear: 2015
  }}
/>
```

Here's a live demo of Gantt Chart on react-handsontable: https://codepen.io/js-ziggle/pen/EXzERP?editors=1000 
(it's a plain html + bundled javascript file, because Handsontable Pro is complicated to configure on codepen/webpackbin/others, due to authentication etc, etc...)
